context-free.md

```diff
@@ -2,21 +2,221 @@
 
 ## What is hard to do in this language? Why is it hard?
 
+It was hard to make images that look like letters -- 
+maybe there is a shortcut I did not realize, but it was a little annoying to 
+have to use lines and arcs to create letters, instead of being able to 
+type out the letters themselves.
 
 ## How did you learn how to program in this language?
 
+I started by modifying the starter file that was an image of "WELCOME" 
+with some vines around it. I changed the location, size, color,
+hue, angle, and saturation of certain lines and shapes, and then started adding
+repeated elements to each object.
+
+I also looked at the documentation site, through which I learned how to add a background color. 
 
 ## What is the underlying _computational model_ for this programming language? 
 _We don't yet have a great definition of the term "computational model". 
 For now, try to come up with the clearest, most concise explanation of what 
 happens when a ContextFree program runs._
 
+I think the program reads the written instructions sequentially, beginning
+with the background color, if it is specified, and moving to the startshape, which is 
+necessary to specify. 
+The program encounters rules, in the order that they are written, and executes all the 
+components of that rule before moving onto the next rule.
 
 ## What do you think is interesting about the ContextFree program you wrote?
 
+I think it's interesting that I found a way to make the stems of the vine
+repeat without explicitly writing and placing each one! 
+
+
+
+
+
+
+
+
+
+# Language
+
+_What is the name of the language? Link the name to its webpage 
+(if appropriate)._
+
+Hibernate Query Language (HQL). Webpage:  
+https://docs.jboss.org/hibernate/orm/3.3/reference/en-US/html/queryhql.html
+
+This link describes Hibernate, which is an object-relational mapping framework 
+for the Java language. It provides a framework for mapping an 
+object-oriented domain model (HQL) to a relational database (SQL).
+http://hibernate.org/
+
+
+# Domain
+_Describe the language's domain in five words._
+
+Persistent objects and their properties.
+
+
+# Computational model
+_We don't yet have a great definition of the term "computational model". 
+For now, try to come up with the clearest, most concise explanation of 
+what happens when a program in your DSL runs._
+
+HQL queries are translated by Hibernate into conventional SQL queries. 
+These SQL queries then perform the action on a database.
+
+
+
+
+# DSL-ness
+_Fowler writes about a spectrum of languages, from general-purpose languages to 
+"purely" domain-specific. Where does the DSL you chose fall on this spectrum, 
+and why?_ 
+
+**I think that HQL falls on the "purely" domain-specific end of the spectrum.**
+
+**HQL definitely fits all the criteria of being a DSL:**
+
+1. Computer Programming Language: 
+HQL is a database independent query language executable by a computer.
```

Comment:
Does it handle NoSQL databases as well? I imagine that would be pretty hard to do (especially efficiently) given the design differences.